src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -726,6 +742,10 @@ private async Task ProcessSubscriptionAsync()
 
                                     AssertCloseWhenNoDocsLeft();
 
+                                    // we might wait for new documents for a long times, lets reduce the stream capacity
+                                    if (_buffer.Capacity > BufferCapacityInBytes)
```

Comment:
setting capacity below length with throw, is this assured here?

Reply:
yes we do `SetLength(0)` on after flushing the subscription batch, in any case I added a debug assert